internal/pkg/agent/application/fleet_server_bootstrap.go

```diff
@@ -85,6 +86,61 @@ func FleetServerComponentModifier(serverCfg *configuration.FleetServerConfig) co
 	}
 }
 
+// ProtectionComponentModifier copies "agent.protection" properties to the top level "protection" for the endpoint input
+// Endpoint uses uninstall_token_hash in order to verify uninstall command token
+// and signing_key in order validate the action signature.
+// Example:
+//
+//	{
+//		....
+//		"protection": {
+//	        "enabled": true,
+//	        "signing_key": "MFkwEwYHKoZIzj0CAQYIKoZIzj0DAQcDQgAEqrEVMJBfAiW7Mz9ZHegwlB7n4deTASUa5LlJlDfuz0hxo/7WPc7gkVB5H8LgnObPfihgzML7rLsHPreWZTB10A==",
+//	        "uninstall_token_hash": ""
+//	    },
+//	    "revision": 1,
+//	    "type": "endpoint"
+//	}
+func ProtectionComponentModifier() coordinator.ComponentsModifier {
```

Comment:
Seems a little odd that this is in the fleet_server_bootstrap file when it has nothing to do with the bootstrap process. I see how it got here, it's where the other modifiers are, but those are injecting the fleet configuration.

This should move somewhere else.

Reply:
renamed